### PR TITLE
Create 2 action buttons inside header component

### PR DESCRIPTION
**Pull request by MillanaCh**
1. I added new package react-icons in package.json and package-lock.json

2. src folder
- app -> changed global.css
Снимок экрана 2023-07-24 в 7 52 20 AM

- app -> page.tsx 
Снимок экрана 2023-07-24 в 7 51 41 AM

**app -> components -> features (new folder)**
- features -> header.tsx (create useState for managing the sidebar modal window)
Снимок экрана 2023-07-24 в 7 52 40 AM

- features -> sidebar.tsx
Снимок экрана 2023-07-26 в 1 43 51 AM

- features -> components -> cities.tsx (created another file for cutting out mapped cities and wrap it in useMemo())
Снимок экрана 2023-07-26 в 1 43 08 AM

- features -> Header.test.tsx (test)
Снимок экрана 2023-07-24 в 7 52 34 AM

### CODE REVIEW

**Review by Alinochka07 — DISMISSED** [review]

**Review by Alinochka07 — DISMISSED** [review]

**Comment by Alex-2kZharkov** [comment]
> 1. I added new package react-icons in package.json and package-lock.json
> 2. src folder
> 
> * app -> changed global.css
> 
> Снимок экрана 2023-07-24 в 7 52 20 AM
> * app -> page.tsx
> 
> Снимок экрана 2023-07-24 в 7 51 41 AM
> **app -> components -> features (new folder)**
> 
> * features -> header.tsx (create useState for managing the sidebar modal window)
> 
> Снимок экрана 2023-07-24 в 7 52 40 AM
> * features -> sidebar.tsx
> 
> Снимок экрана 2023-07-26 в 1 43 51 AM
> * features -> components -> cities.tsx (created another file for cutting out mapped cities and wrap it in useMemo())
> 
> Снимок экрана 2023-07-26 в 1 43 08 AM
> * features -> Header.test.tsx (test)
> 
> Снимок экрана 2023-07-24 в 7 52 34 AM

I really liked you give such a detailed description. If takes a long time, you can skip screenshots. But still: the description is good. Keep it the same way in future when you work real job

File: .eslintrc.json [hunk 1]
```diff
@@ -9,13 +9,15 @@
...
     "func-names": "off",
     "no-process-exit": "off",
     "class-methods-use-this": "off",
     "import/prefer-default-export": "off",
     "react/react-in-jsx-scope": "off",
     "react/function-component-definition": "off",
-    "no-param-reassign": "warn"
+    "jsx-a11y/click-events-have-key-events": "off",
+    "jsx-a11y/no-static-element-interactions": "off",
+    "no-param-reassign": "off", //did it temporarily
```
Comment (Alex-2kZharkov) [inline .eslintrc.json]: set both of them to `warn` forever

File: src/components/features/Header.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const setIsOpen = jest.fn();
+
+describe('HeaderMainPage', () => {
+  test('pass the header main page buttons', () => {
+    render(<HeaderPage />);
+
+    const btnRecord = screen.getByRole('button', {
+      name: 'запись онлайн',
+    });
+    userEvent.click(btnRecord);
```
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx]: it is an async thing, use `await userEvent.click`

File: src/components/features/Header.test.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    expect(btnRecord).toBeEnabled();
```
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx]: empty line please

File: src/components/features/Header.test.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    const btnBuy = screen.getByRole('button', {
```
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx]: use `await findByRole` if this thing should appear after you click on button

Comment (Alex-2kZharkov) [reply src/components/features/Header.test.tsx]: +1 empty line.

Overall good tests

File: src/components/features/Header.test.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      name: 'купить косметику',
+    });
+    expect(btnBuy).toBeEnabled();
+  });
+
+  test('pass the sidebar test', () => {
+    render(<Sidebar isOpen={isOpen} setIsOpen={setIsOpen} />);
```
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx]: wait a minute. tests for `Sidebar` should be in the `sidebar.test.tsx` file

File: src/components/features/Header.test.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    // Search bar
```
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx]: remove comment, no comments for such simple things. You have descriptive `searchBox` variable

File: src/components/features/Header.test.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    const searchBox = screen.getByPlaceholderText('Найти города', { exact: false });
+    expect(searchBox).toBeInTheDocument();
```
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx]: +1 empty line

File: src/components/features/components/cities.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Alex-2kZharkov) [inline src/components/features/components/cities.tsx]: you should set correct options in eslintrc, see https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md

Comment (Alex-2kZharkov) [reply src/components/features/components/cities.tsx]: and remove comment in the end

File: src/components/features/components/cities.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  { city: 'Воронеж', branch: 4 },
+  { city: 'Долгопрудный', branch: 3 },
+  { city: 'Красноярск', branch: 3 },
+  { city: 'Раменское', branch: 3 },
+];
+
+export function Location(): ReactElement {
+  // for later using onClick to naviagte to the locations
+
+  // const handleCityClick = useCallback(
```
Comment (Alex-2kZharkov) [inline src/components/features/components/cities.tsx]: actually, better to uncomment and use it right now with console.log

File: src/components/features/components/cities.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  //     console.log('City Show');
+  //   },
+  //   [],
+  // );
+
+  return useMemo(
+    () => (
+      <ul>
+        {cityLocations.map(({ city, branch }) => (
+          // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
```
Comment (Alex-2kZharkov) [inline src/components/features/components/cities.tsx]: I thought we have turned it off in eslintrc , if no, please turn it off

File: src/components/features/components/cities.tsx
Comment (Alex-2kZharkov) [inline src/components/features/components/cities.tsx:1]: index.ts is required in this folder. It is a main rule🙂

File: src/components/features/header.tsx
Comment (Alex-2kZharkov) [inline src/components/features/header.tsx:1]: no-no, it should be `src/features` not `src/components/features`

File: src/components/features/header.tsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+import Link from 'next/link';
+import { useState } from 'react';
+
+import { Header } from '../Header';
+import { PromoBanner } from '../PromoBanner';
+import { barbsershopData } from '../PromoBanner/PromoData';
+import { Sidebar } from './sidebar';
+
+export function HeaderPage(): JSX.Element {
```
Comment (Alex-2kZharkov) [inline src/components/features/header.tsx]: arrow function please. It is your default syntax

File: src/components/features/header.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <Header />
+      <PromoBanner barbershopCount={barbsershopData.barbershopCount} />
+      {isOpen ? <Sidebar isOpen={isOpen} setIsOpen={setIsOpen} /> : <div />}
```
Comment (Alex-2kZharkov) [inline src/components/features/header.tsx]: empty div looks really strange, better `isOpen && <Sidebar />`

File: src/components/features/header.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+      <div className="display: inline-grid absolute top-96 mt-44 ml-12">
+        <button
+          onClick={() => setIsOpen(!isOpen)}
```
Comment (Alex-2kZharkov) [inline src/components/features/header.tsx]: please read this and fix accordingly https://www.delftstack.com/howto/react/react-setstate-prevstate/

File: src/components/features/Header.test.tsx
Comment (Alex-2kZharkov) [inline src/components/features/Header.test.tsx:1]: the component is written in lower case, so this file also `header.test.tsx`

File: src/components/features/components/cities.tsx
Comment (Alex-2kZharkov) [inline src/components/features/components/cities.tsx:1]: create a separate folder `cities` here and put this file there

File: src/components/features/sidebar.tsx
Comment (Alex-2kZharkov) [inline src/components/features/sidebar.tsx:1]: create a separate folder `sidebar` here and put this file there

File: src/components/features/sidebar.tsx [hunk 14]
```diff
@@ -0,0 +1,82 @@
+import { Location } from './components/cities';
+
+interface ModalWindow {
```
Comment (Alex-2kZharkov) [inline src/components/features/sidebar.tsx]: catch you. Why you name if ModalWindow if you apply it to the Sidebar? it should be SidebarProps

File: src/components/features/sidebar.tsx [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  isOpen: boolean;
+  setIsOpen: (value: boolean) => void;
```
Comment (Alex-2kZharkov) [inline src/components/features/sidebar.tsx]: good usage of TS, keep the good work

File: src/components/features/components/cities.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+  { city: 'Санкт-Петербург', branch: 16 },
+  { city: 'Калининград', branch: 10 },
+  { city: 'Красногорск', branch: 5 },
+  { city: 'Воронеж', branch: 4 },
+  { city: 'Долгопрудный', branch: 3 },
+  { city: 'Красноярск', branch: 3 },
+  { city: 'Раменское', branch: 3 },
+];
+
+export function Location(): ReactElement {
```
Comment (Alex-2kZharkov) [inline src/components/features/components/cities.tsx]: one more time: this component name and file name should match. Currently they do not: `cities.tsx` and `Location`. Please, select one and fix

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: src/Features/sidebar/sidebar.tsx
Comment (Alex-2kZharkov) [inline src/Features/sidebar/sidebar.tsx:1]: index.ts for sidebar folder🙂

**Review by Alex-2kZharkov — DISMISSED** [review]

**Review by Alex-2kZharkov — DISMISSED** [review]

File: src/Features/header/header.test.tsx [hunk 17]
```diff
@@ -12,13 +12,12 @@ describe('HeaderMainPage', () => {
...
-    const btnRecord = screen.getByRole('button', { name: 'запись онлайн' });
+    const btnRecord = screen.getByRole('button', { name: 'Запись онлайн' });
     await userEvent.click(btnRecord);
 
     expect(btnRecord).toBeEnabled();
 
-    const btnBuy = await screen.findByRole('button', { name: 'купить косметику' });
-
-    expect(btnBuy).toBeEnabled();
+    const btnBuy = screen.getByText('Купить косметику');
```
Comment (Alex-2kZharkov) [inline src/Features/header/header.test.tsx:20]: should be full name `buttonName` + 1 empty line between expect and `const`

**Review by Alex-2kZharkov — APPROVED** [review]

**Review by Alex-2kZharkov — APPROVED** [review]

**Review by Alex-2kZharkov — APPROVED** [review]

**Pull request merged**